docs/types/value-types.rst

```diff
@@ -494,21 +496,22 @@ regardless of the type of the right (exponent) operand.
     Division on integer literals used to truncate in Solidity prior to version 0.4.0, but it now converts into a rational number, i.e. ``5 / 2`` is not equal to ``2``, but to ``2.5``.
 
 .. note::
-    Solidity has a number literal type for each rational number.
-    Integer literals and rational number literals belong to number literal types.
+    In solidity, rational numbers and integers expressed as literals are of number literal type.
```

Comment:
Your version as a different meaning. The original sentence says that e.g. `1` and `2` belong to different types (called `int_const 1` and `int_const 2`; you can't use these types for variables but you can sometimes see them in error messages), while `1` and `1.0` are of the same type (both are `int_const 1`).

The important takeaway is that there are no separate types for integers and fractional numbers used in literals. In the AST `1.0` is indistinguishable from `1`. That distinction only appears when you convert the literal expression to a concrete integer or fixed-point type (note also that fixed-point types are currently not fully implemented).